### PR TITLE
Refactor PF.AppChooserDialog

### LINKED ISSUE #554: Open in --> Other Application... has wrong headerbar
## Actual behavior

screenshot from 2018-07-21 18-59-54

## Expected behavior

screenshot from 2018-07-21 18-59-54-2

## Steps to reproduce the behavior

1. Open Files
2. Right-click and select "Open in" --> "Other Application..."
3. The dialog has wrong headerbar

I'm not sure this issue is whether a bug or intentional feature.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/61158303-open-in-other-application-has-wrong-headerbar?utm_campaign=plugin&utm_content=tracker%2F65602118&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F65602118&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by jeremypw**
PF.AppChooserDialog now derived from Gtk.Dialog, fixing the linked issue and allowing more flexibility in design.

When showing unrelated applications the set as default box is unchecked to avoid accidentally setting an unsuitable app as default for that type.

Required content type is derived from pre-existing GOF.File rather than querying again.

### CODE REVIEW

File: src/Dialogs/ChooseAppDialog.vala [hunk 1]
```diff
@@ -19,51 +19,100 @@
...
+        show_switch.active = false;
+        show_switch.state = false;
+
+        show_grid.add (show_label);
+        show_grid.add (show_switch);
+
+        grid.add (show_grid);
+        content_area.add (grid);
+
+        add_button (_("Select"), Gtk.ResponseType.OK);
```
Comment (ryonakano) [inline src/Dialogs/ChooseAppDialog.vala]: I wonder why this is not `add_button (_("_Select"), Gtk.ResponseType.OK);`.

File: src/Dialogs/ChooseAppDialog.vala [hunk 2]
```diff
@@ -19,51 +19,100 @@
...
+        add_button (_("Cancel"), Gtk.ResponseType.CANCEL);
```
Comment (ryonakano) [inline src/Dialogs/ChooseAppDialog.vala]: Same as above, `add_button (_("_Cancel"), Gtk.ResponseType.CANCEL);`

**Comment by jeremypw** [comment]
Elementary apps tend not to have button mnemonics - not sure why - I'll check with the UX team.

**Comment by jeremypw** [comment]
In this case the buttons are keyboard accessible anyway - use the arrow and Enter keys. There may be an argument for making the checkbox and switch keyboard accessible.

**Comment by danrabbit** [comment]
For reference:

screenshot from 2018-07-22 10 30 44 2x

**Comment by jeremypw** [comment]
Thanks for the feedback.  

What do you suggest instead of the switch?  I did not like the large button much.

I must admit I did not realise the search bar existed!  Is this what causes the horizontal line that was the subject of the issue?

**Comment by danrabbit** [comment]
I'm not sure. I think I kind of prefer the big button to the switch but I'm not sure what would be even better than that

Yeah the search bar is where the horizontal line comes from. I wonder if it might be easier to try to get rid of this from the stylesheet side and we can consider the overall design of this dialog some other time as part of a Granite widget or something.

**Comment by jeremypw** [comment]
@ryonakano UX team say button mnemonics are not desirable at the moment because of translation implications. They may be considered in the future.

**Comment by jeremypw** [comment]
It seems that Gtk.AppChooserDialog uses a function of the widget to link in the search entry which is kindly not exposed publicly.  Therefore there is no simple way to reproduce this functionality at the moment.  It would be necessary to write a new AppChooser widget which is not really worthwhile.  I'll leave this open for a while in case someone knows different.

**Pull request closed without merging**